src/comparsion.py

```diff
@@ -0,0 +1,161 @@
+            
+            # These lines convert these columns to string format
+            p["CQL_Code"] = p["CQL_Code"].apply(lambda x: '\n'.join(x) if isinstance(x, list) else x)
+            p["File"] = p["File"].apply(lambda x: '\n'.join(x) if isinstance(x, list) else x)
+            p["File"] = p["File"].apply(lambda x: re.sub(codeql_system_path, '', x, count=1))
+            p["CQL_Column"] = p["CQL_Column"].astype(float)
+            dataframes[dataframe_key] = p
+
+    if whichtwo == "CLJvCNL" or whichtwo == "CLJvCQL" or whichtwo == "all":
+    # Load Clojure dataset and print the atoms
+        clojure_master = pd.read_csv(clojure_directory)
+        clojure_master["file"] = clojure_master["file"].apply(lambda x: re.sub(clojure_system_path, '', x, count=1))
+
+    # Prepare Clojure dataset for merge
+
+        clojure_master['atom'] = clojure_master['atom'].map(combined_mapping)
+        clojure_master.rename(columns = {
+            'atom': 'Type',
+            'file': 'File',
+            'line': 'Line',
+            'code': 'CLJ_Code'
+        }, inplace=True)
+
+        clojure_master["CLJ"] = '+'
+
+        clojure_master.drop(columns = ['offset'], inplace = True)
+
+    # Merge Clojure and CodeQL dataframes
+        for key, df in dataframes.items():
+            k = pd.merge(clojure_master[clojure_master["Type"] == key], df, on = ['File', 'Type', 'Line'], how = 'outer')
+            k['CLJ'] = k['CLJ'].fillna('-')
+            dataframes[key] = k.sort_values(by = ["File", "Line"]).reset_index().drop("index", axis = 1)
+
+    # Load Coccinelle dataframes
+
+    if whichtwo == "CQLvCNL" or whichtwo == "CLJvCNL" or whichtwo == "all":
+        coccinelle_dataframes = {}
+        for file_path in Path(coccinelle_directory).glob('*.csv'):
+            filename = Path(file_path).stem
+            dataframe_key = filename
+            p = pd.read_csv(file_path)
+            p.columns = ["Type", "File", "Line", "CNL_Column", "CNL_Code"]
+            p["File"] = p["File"].apply(lambda x: re.sub(coccinelle_system_path, '', x, count=1))
+            p["CNL"] = '+'
+            coccinelle_dataframes[dataframe_key] = p
+
+        for i in coccinelle_dataframes.values():
+            i['Type'] = i['Type'].map(combined_mapping)
+
+        # Merge Coccinelle DataFrames
+
+        for i in coccinelle_dataframes.values():
+            key = i.loc[0, "Type"]
+            toMerge = dataframes[key]
+            k = pd.merge(toMerge, i, on = ['File', 'Type', 'Line'], how = 'outer')
+            k['CNL'] = k['CNL'].fillna('-')
+            dataframes[key] = k.sort_values(by = ["File", "Line"]).reset_index().drop("index", axis = 1)
+
+    # Output merged CSVs to output_directory
+
+    for key, df in dataframes.items():
+        df = df.reindex(columns=["Type", "File", "Line", "CLJ", "CQL", "CNL", "CLJ_Code", "CQL_Column", "CQL_Code", "CNL_Column", "CNL_Code"])
+        df = pd.DataFrame(df)
+        if whichtwo == "CLJvCQL":
```

Comment:
If you need an if with elif and no else, it's better to simply write multiple if statements. Additionally, if you ever need to define something with an empty body, you can use `pass`